lib/ab_test.rb

```diff
@@ -36,25 +39,44 @@ def initialize(
   # @params [Hash] session
   # @param [User] user
   # @param [Hash] user_session
-  def bucket(request:, service_provider:, session:, user:, user_session:)
-    return nil if no_percentages?
-
+  # @param [Boolean] persisted_read_only Avoid new bucket assignment if test is configured to be
+  # persisted but there is no persisted value.
+  def bucket(
+    request:,
+    service_provider:,
+    session:,
+    user:,
+    user_session:,
+    persisted_read_only: false
+  )
     discriminator = resolve_discriminator(
       request:, service_provider:, session:, user:,
       user_session:
     )
     return nil if discriminator.blank?
 
+    persisted_value = AbTestAssignment.bucket(experiment:, discriminator:) if persist
+    return persisted_value if persisted_value || (persist && persisted_read_only)
+
+    return nil if no_percentages?
```

Comment:
wouldn't we want to return `@default_bucket` here?

Reply:
Interesting question. This specific code existed previously, I had just rearranged it. But I'm not sure if `nil` is meant to be some meaningful value separate from `@default_bucket`. @matthinz Any thoughts here, since you recently worked on this?